src/prefect/client/schemas/objects.py

```diff
@@ -305,6 +305,9 @@ def is_cancelled(self) -> bool:
     def is_cancelling(self) -> bool:
         return self.type == StateType.CANCELLING
 
+    def is_cached(self) -> bool:
+        return self.name == "Cached" and self.type == StateType.COMPLETED
+
```

Comment:
sorry to retroactively request a nit!

but can we remove and inline this? just because this method will not make sense in 3.x